packages/block-library/src/separator/style.scss

```diff
@@ -8,6 +8,13 @@
 	border-bottom: none;
 }
 
+// 0-2-0 specificity required to enforce block instance
+// color selections over global styles.
+.wp-block-separator.has-background,
+.wp-block-separator.has-text-color {
+	border-color: currentColor;
+}
```

Comment:
Just so I understand:

- if the separator block has a background color set, the `color` property will have the same value (thanks to the changes in this PR),
- this rule therefore ensures the border color has the same value as `color`, through `currentColor`

> Gave this a test. In TT4, I'm seeing that the block's instance color settings aren't overriding global style color for the 'Default' or 'Wide Line' block styles variations:

The CSS rule above addresses this issue for me in the editor and frontend:

1. In TT4, change the background value of Wide Line variation in global styles
2. At the block level, apply Wide Line variation, and overwrite the background color

Do block style variation global styles need the same treatment? 🤔 

1. In TT4, change the background value of Wide Line variation in global styles
2. At the block level, apply Wide Line variation

The background color is set, but the [theme.json border color](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentyfour/theme.json#L740) still affects the appearance. Here's what I'm seeing in in the frontend:

```css
:root :where(.wp-block-separator.is-style-wide--2) {
    background-color: #00ff40;
}

:root :where(.wp-block-separator) {
}
:root :where(.wp-block-separator) {
    border-color: currentColor;
    border-width: 0 0 1px 0;
    border-style: solid;
    color: var(--wp--preset--color--contrast);
}
```